core/arch/x86/decode.c

```diff
@@ -1828,6 +1837,40 @@ decode_operand(decode_info_t *di, byte optype, opnd_size_t opsize, opnd_t *opnd)
         /* no need to set size as it's a GPR */
         return true;
     }
+    case TYPE_K_MODRM: {
+        /* part of AVX-512: modr/m.rm selects opmask register or mem addr */
+        if (di->mod != 3) {
+            return decode_modrm(di, optype, opsize, NULL, opnd);
+        }
+        /* fall through*/
+    }
+    case TYPE_K_MODRM_R: {
+        /* part of AVX-512: modr/m.rm selects opmask register */
+        *opnd = opnd_create_reg(decode_reg(DECODE_REG_RM, di, optype, opsize));
+        /* For VEX encoded scalar opmask instructions, the size is determined
+         * by the opcode.
+         */
+        return true;
+    }
+    case TYPE_K_REG: {
+        /* part of AVX-512: modr/m.reg selects opmask register */
+        *opnd = opnd_create_reg(decode_reg(DECODE_REG_REG, di, optype, opsize));
+        /* AVX-512 VEX encoded scalar opmask instructions, the size is determined
+         * by the opcode.
+         */
+        return true;
+    }
+    case TYPE_K_VEX: {
+        /* part of AVX-512: vex.vvvv selects opmask register */
+        *opnd = opnd_create_reg(decode_reg(DECODE_REG_VEX, di, optype, opsize));
+        /* AVX-512 VEX encoded scalar opmask instructions, the size is determined
+         * by the opcode.
+         */
+        return true;
+    }
+    case TYPE_K_EVEX: {
+        CLIENT_ASSERT(false, "decode error: unsupported.");
```

Comment:
As you know an assert like this should not be used for a legitimate code path.  If this is not yet implemented as part of incremental code it needs clear labeling as such in a comment and the output message.

Reply:
Done.